BUG: sample returning a different size in pymc 5.7.0

### Describe the issue:

With pymc == 5.6.1 I get:
```
In [2]: import pymc
WARNING (pytensor.tensor.blas): Using NumPy C-API based implementation for BLAS functions.

In [3]: with pymc.Model():
   ...:     pymc.Bernoulli('X', p=0.2)
   ...:     samps = pymc.sample(draws=3, chains=1, return_inferencedata=False)
   ...: 
Only 3 samples in chain.
Sequential sampling (1 chains in 1 job)
BinaryGibbsMetropolis: [X]
Sampling 1 chain for 1_000 tune and 3 draw iterations (1_000 + 3 draws total) took 0 seconds.rgences]
The number of samples is too small to check convergence reliably.

In [4]: samps[:]['X'].shape
Out[4]: (3,)
```
However with pymc==5.7.0 I get:
```
In [1]: import pymc
WARNING (pytensor.tensor.blas): Using NumPy C-API based implementation for BLAS functions.

In [2]: with pymc.Model():
   ...:     pymc.Bernoulli('X', p=0.2)
   ...:     samps = pymc.sample(draws=3, chains=1, return_inferencedata=False)
   ...: 
Only 3 samples in chain.
Sequential sampling (1 chains in 1 job)
BinaryGibbsMetropolis: [X]
Sampling 1 chain for 0 tune and 1_003 draw iterations (0 + 1_003 draws total) took 0 seconds.rgences]
Only one chain was sampled, this makes it impossible to run some convergence checks

In [3]: samps[:]['X'].shape
Out[3]: (1003,)
```
So the shape of the generated output has changed.

### Reproduceable code example:

```python
import pymc
with pymc.Model():
    pymc.Bernoulli('X', p=0.2)
    samps = pymc.sample(draws=3, chains=1, return_inferencedata=False)
print(samps[:]['X'].shape)
```


### Error message:

_No response_

### PyMC version information:

5.7.0

### Context for the issue:

This is just coming up in SymPy CI. From a SymPy VCS checkout this is reproduced with:
```
$ pytest sympy/stats/sampling/tests/test_sample_finite_rv.py -k test_sample_pymc
...
            with pymc.Model():
                if do_sample_pymc(self):
                    samps = pymc.sample(draws=prod(size), chains=1, compute_convergence_checks=False,
                            progressbar=False, random_seed=seed, return_inferencedata=False)[:]['X']
>                   samps = samps.reshape(size)
E                   ValueError: cannot reshape array of size 1003 into shape (1,3)

sympy/stats/rv.py:1639: ValueError
```
The code in question is here:
https://github.com/sympy/sympy/blob/e8ffcf40a6148235667ed75b811bdc1f953466e9/sympy/stats/rv.py#L1635-L1639

--- a/pymc/step_methods/metropolis.py
+++ b/pymc/step_methods/metropolis.py
@@ -492,7 +492,6 @@ class BinaryGibbsMetropolis(ArrayStep):
 
     def reset_tuning(self):
         # There are no tuning parameters in this step method.
-        self.tune = False
         return
 
     def astep(self, apoint: RaveledVars, *args) -> Tuple[RaveledVars, StatsType]:
@@ -620,7 +619,6 @@ class CategoricalGibbsMetropolis(ArrayStep):
 
     def reset_tuning(self):
         # There are no tuning parameters in this step method.
-        self.tune = False
         return
 
     def astep_unif(self, apoint: RaveledVars, *args) -> Tuple[RaveledVars, StatsType]:

--- a/tests/helpers.py
+++ b/tests/helpers.py
@@ -133,7 +133,7 @@ class StepMethodTester:
                     continue
                 assert idata.sample_stats[stat].dtype == np.dtype(dtype)
 
-    def step_continuous(self, step_fn, draws):
+    def step_continuous(self, step_fn, draws, chains=1, tune=1000):
         start, model, (mu, C) = mv_simple()
         unc = np.diag(C) ** 0.5
         check = (("x", np.mean, mu, unc / 10), ("x", np.std, unc, unc / 10))
@@ -143,14 +143,19 @@ class StepMethodTester:
             with warnings.catch_warnings():
                 warnings.filterwarnings("ignore", "More chains .* than draws .*", UserWarning)
                 idata = pm.sample(
-                    tune=1000,
+                    tune=tune,
                     draws=draws,
-                    chains=1,
+                    chains=chains,
                     step=step,
                     initvals=start,
                     model=model,
                     random_seed=1,
+                    discard_tuned_samples=False,
                 )
+            assert idata.warmup_posterior.sizes["chain"] == chains
+            assert idata.warmup_posterior.sizes["draw"] == tune
+            assert idata.posterior.sizes["chain"] == chains
+            assert idata.posterior.sizes["draw"] == draws
             self.check_stat(check, idata, step.__class__.__name__)
             self.check_stat_dtype(idata, step)
 

